src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/DomainEventService.kt

```diff
@@ -127,31 +121,27 @@ class DomainEventService(
     saveAndEmit(
       domainEvent = domainEvent,
       eventType = DomainEventType.APPROVED_PREMISES_BOOKING_NOT_MADE,
-      nomsNumber = domainEvent.data.eventDetails.personReference.noms,
```

Comment:
I do wonder now if we can remove the event type specific calls now and just delegate to `saveAndEmit` and work out the eventType from there?

Reply:
I guess using a reified generic somehow? sounds like a good idea

Reply:
actually probably easiest to just add type into DomainEvent :) Yes, then can remove a bunch of functions.